lib/importers/licence_transaction/licence_importer.rb

```diff
@@ -80,11 +85,35 @@ def licences
         )["results"]
       end
 
-      def filtered_licences
-        licence_identifiers = File.new(common_licences_path).readlines(chomp: true)
+      def licences_tagging
+        @licences_tagging ||= grouped_licences.map do |licence_url, rows|
+          {
+            "base_path" => URI.parse(licence_url).path,
+            "locations" => locations(rows).uniq,
+            "industries" => industries(rows).uniq,
+          }
+        end
+      end
+
+      def tags_for_licence(base_path)
+        licences_tagging.find { |l| l["base_path"] == base_path }
+      end
 
-        licences.select do |licence|
-          licence_identifiers.include?(licence["details"]["licence_identifier"])
+      def locations(grouped_licence)
+        grouped_licence.flat_map do |licence|
+          licence["Locations"].split(", ").map(&:parameterize)
+        end
+      end
+
+      def industries(grouped_licence)
+        grouped_licence.flat_map do |licence|
+          licence[2..].compact.map(&:parameterize)
```

Comment:
We're already are [here](https://github.com/alphagov/specialist-publisher/pull/2269/files/660f5aa5a619ba781086be015ecb0198bd7271f4..720620a1b69ac05349a40a5f5464d14a219099aa#diff-9fefdf86044e4f7f4075d0f74f7b114aecd6316621ac1cc8ba0c17621699e39aR92-R93). Having a unique at that level ensures that there are no unique values in the final list (vs here where it would only ensure uniqueness within each row).